src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,1296 @@
+{
+    void M()
+    {
+        string s = ""Test"";
+        throw new Exception(s.ToUpper($$
+
+    }
+}";
+
+            var expected = @"
+public class Class1
+{
+    void M()
+    {
+        string s = ""Test"";
+        throw new Exception(s.ToUpper());$$
+
+    }
+}";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfNestedMethodInvocation1_MissingOuterParen()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y.ToString($$)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y.ToString());$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfNestedMethodInvocation2_MissingBothParens()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y.ToString($$");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y.ToString());$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfNestedMethodInvocation3_DualPosition_MissingOuter()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString($$), y");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString(), y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfNestedMethodInvocation3_DualPosition()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString($$), y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString(), y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfNestedMethodInvocation3_DualPosition2()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString($$), y);");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString(), y);;$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfNestedMethodInvocation4_knownfailure()
+        {
+            // Because of the open paren, thinks everything after is the first argument
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString($$, y");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x.ToString(), y);$$");
```

Comment:
:thought_balloon: I think we can get away with the current behavior for now, given most users will have automatic brace completion enabled to save them from this case.